### PR TITLE
feat(social-insurance-administration): Add e2e tests

### PR DESCRIPTION
**Pull request by veronikasif**
[[TS-106] e2e tests (Old-age-pension)](https://dit-iceland.atlassian.net/jira/software/c/projects/TS/boards/82?selectedIssue=TS-106)
[[TS-319] e2e tests (Additional-support-for-the-elderly)](https://dit-iceland.atlassian.net/browse/TS-319)
[[TS-156] e2e tests (Pension-supplement)](https://dit-iceland.atlassian.net/jira/software/c/projects/TS/boards/82?selectedIssue=TS-156)
[[TS-155] e2e tests (Household-supplement)](https://dit-iceland.atlassian.net/jira/software/c/projects/TS/boards/82?selectedIssue=TS-155)
[[TS-324] e2e test for 1/2 old age pension](https://dit-iceland.atlassian.net/jira/software/c/projects/TS/boards/82?selectedIssue=TS-324)

## What

Added e2e tests for TR applications

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Added mock responses for National Registry, Parental Leave, and Social Insurance Administration services to enhance testing capabilities.
	- Introduced end-to-end test suites for the application processes related to Additional Support for the Elderly, Household Supplement, Pension Supplement, and Old Age Pension.
	- Enhanced identification for testing purposes with new `dataTestId` properties in various forms.
	- Improved form interactivity with dynamic options based on user input for payment periods and bank account types.
	- Added new utility functions to streamline user interactions within forms.

- **Bug Fixes**
	- Improved mock loading strategy for better organization and efficiency.

### CODE REVIEW

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/setup-xroad.mocks.ts [hunk 1]
```diff
@@ -1,288 +1,20 @@
...
-import addMonths from 'date-fns/addMonths'
 import { getEnvVariables } from '../../../../../../../infra/src/dsl/service-to-environment/pre-process-service'
-import { env } from '../../../../support/urls'
 import { EnvironmentConfig } from '../../../../../../../infra/src/dsl/types/charts'
+import { Base } from '../../../../../../../infra/src/dsl/xroad'
+import { env } from '../../../../support/urls'
+import { resetMocks, wildcard } from '../../../../support/wire-mocks'
+import { loadParentalLeaveXroadMocks } from './mocks/parentalLeave.mock'
+import { loadSocialInsuranceAdministrationXroadMocks } from './mocks/socialInsuranceAdministration.mock'
+import { loadNationalRegistryXroadMocks } from './mocks/nationalRegistry.mocks'
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/setup-xroad.mocks.ts:8]: Fixed

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/old-age-pension.spec.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    async ({ applicationPage }) => {
+      const page = applicationPage
+      const { proceed } = helpers(page)
+
+      await applicationTest.step('Select type of application', async () => {
+        await expect(
+          page.getByRole('heading', {
+            name: label(oldAgePensionFormMessage.pre.applicationTypeTitle),
+          }),
+        ).toBeVisible()
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/old-age-pension.spec.ts:49]: Ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/nationalRegistry.mock.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  await addXroadMock({
+    config: NationalRegistry,
+    prefix: 'XROAD_NATIONAL_REGISTRY_SERVICE_PATH',
+    apiPath: '/api/v1/lyklar/hjuskaparkodar/1/1',
+    prefixType: 'only-base-path',
+    response: new Response().withJSONBody({
+      kodi: '1',
+      lysing: 'Ógiftur',
+    }),
+  })
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/nationalRegistry.mock.ts:161]: Not needed, ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/additional-support-for-the-elderly.spec.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            }),
+        ).toBeVisible()
+        await page
+          .getByRole('button', {
+            name: label(
+              socialInsuranceAdministrationMessage.confirm.submitButton,
+            ),
+          })
+          .click()
+      })
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/additional-support-for-the-elderly.spec.ts:193]: This is done in the 'Check that conclusion screen header is visible' step below

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/parentalLeave.mock.ts [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+            sveitarfelagsnumer: '1000',
+          },
+          adsetur: {
+            heiti: 'Fellsmúli 2',
+            postnumer: '108',
+            stadur: 'Reykjavík',
+            sveitarfelagsnumer: '0000',
+          },
+        },
+      ],
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/parentalLeave.mock.ts:202]: Ok as is

File: libs/application/templates/social-insurance-administration/additional-support-for-the-elderly/src/forms/AdditionalSupportForTheElderlyForm.ts [hunk 6]
```diff
@@ -287,6 +288,7 @@ export const AdditionalSupportForTheElderlyForm: Form = buildForm({
                     socialInsuranceAdministrationMessage.payment
                       .personalAllowancePercentage,
                   suffix: '%',
+                  dataTestId: 'personal-allowance-usage',
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply libs/application/templates/social-insurance-administration/additional-support-for-the-elderly/src/forms/AdditionalSupportForTheElderlyForm.ts:291]: ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/nationalRegistry.mock.ts [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    config: NationalRegistry,
+    prefix: 'XROAD_NATIONAL_REGISTRY_SERVICE_PATH',
+    apiPath: '/api/v1/einstaklingar/0101307789/faedingarstadur',
+    prefixType: 'only-base-path',
+    response: new Response().withJSONBody({
+      sveitarfelagsnumer: '0000',
+      stadur: 'Reykjavík',
+      faedingardagur: '1930-01-01T00:00:00',
+    }),
+  })
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/nationalRegistry.mock.ts:156]: Ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/nationalRegistry.mock.ts [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+      },
+    ]),
+  })
+  await addXroadMock({
+    config: NationalRegistry,
+    prefix: 'XROAD_NATIONAL_REGISTRY_SERVICE_PATH',
+    apiPath: '/api/v1/einstaklingar/0101303019/samibuar',
+    prefixType: 'only-base-path',
+    response: new Response().withJSONBody(['0101303019']),
+  })
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/mocks/nationalRegistry.mock.ts:92]: Ok as is

File: libs/application/templates/social-insurance-administration/pension-supplement/src/forms/PensionSupplementForm.ts [hunk 9]
```diff
@@ -312,7 +313,24 @@ export const PensionSupplementForm: Form = buildForm({
...
               placeholder:
                 socialInsuranceAdministrationMessage.period.monthDefaultText,
-              options: MONTHS,
+              options: (application: Application) => {
+                const { selectedYear } = getApplicationAnswers(
+                  application.answers,
+                )
+
+                return getAvailableMonths(selectedYear)
+              },
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply libs/application/templates/social-insurance-administration/pension-supplement/src/forms/PensionSupplementForm.ts:322]: Ok as is

File: libs/application/templates/social-insurance-administration/household-supplement/src/forms/HouseholdSupplementForm.ts [hunk 10]
```diff
@@ -530,7 +531,24 @@ export const HouseholdSupplementForm: Form = buildForm({
...
+                const { selectedYear, selectedYearHiddenInput } =
+                  getApplicationAnswers(answers)
+
+                return selectedYear === selectedYearHiddenInput
+              },
+            }),
+            buildHiddenInputWithWatchedValue({
+              // Needed to trigger an update on options in the select above
+              id: 'period.hiddenInput',
+              watchValue: 'period.year',
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply libs/application/templates/social-insurance-administration/household-supplement/src/forms/HouseholdSupplementForm.ts:551]: Ok as is

File: libs/application/templates/social-insurance-administration/pension-supplement/src/forms/PensionSupplementForm.ts [hunk 11]
```diff
@@ -312,7 +313,24 @@ export const PensionSupplementForm: Form = buildForm({
...
+                const { selectedYear, selectedYearHiddenInput } =
+                  getApplicationAnswers(answers)
+
+                return selectedYear === selectedYearHiddenInput
+              },
+            }),
+            buildHiddenInputWithWatchedValue({
+              // Needed to trigger an update on options in the select above
+              id: 'period.hiddenInput',
+              watchValue: 'period.year',
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply libs/application/templates/social-insurance-administration/pension-supplement/src/forms/PensionSupplementForm.ts:333]: Ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+  await page
+    .getByPlaceholder(
+      label(
+        socialInsuranceAdministrationMessage.additionalInfo.commentPlaceholder,
+      ),
+    )
+    .fill(
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vehicula malesuada augue, sit amet pulvinar tortor pellentesque at. Nulla facilisi. Nunc vel mi ac mi commodo rhoncus sit amet ut neque.',
+    )
+  await proceed()
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts:119]: Ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+    }),
+  ).toBeVisible()
+
+  await page.getByTestId('select-period.year').click()
+  await page.keyboard.press('Enter')
+
+  await page.getByTestId('select-period.month').click()
+  await page.keyboard.press('ArrowUp')
+  await page.keyboard.press('Enter')
+  await proceed()
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts:84]: It does not work, getting error: `page.selectOption: Error: Element is not a <select> element`.
Ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+        socialInsuranceAdministrationMessage.info.infoSubSectionTitle,
+      ),
+    }),
+  ).toBeVisible()
+
+  const phoneNumber = page.getByRole('textbox', {
+    name: label(socialInsuranceAdministrationMessage.info.applicantPhonenumber),
+  })
+  await phoneNumber.selectText()
+  await phoneNumber.fill('6555555')
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts:21]: Ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+  await expect(
+    page.getByRole('heading', {
+      name: label(socialInsuranceAdministrationMessage.payment.title),
+    }),
+  ).toBeVisible()
+  const paymentBank = page.getByRole('textbox', {
+    name: label(socialInsuranceAdministrationMessage.payment.bank),
+  })
+  await paymentBank.selectText()
+  await paymentBank.fill('051226054678')
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts:37]: ok as is

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+  const { proceed } = helpers(page)
+
+  await expect(
+    page.getByRole('heading', {
+      name: label(
+        socialInsuranceAdministrationMessage.info.infoSubSectionTitle,
+      ),
+    }),
+  ).toBeVisible()
+
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts]: Fixed

File: apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+      .getByText(
+        label(socialInsuranceAdministrationMessage.confirm.overviewTitle),
+      ),
+  ).toBeVisible()
+  await page
+    .getByRole('button', {
+      name: label(socialInsuranceAdministrationMessage.confirm.submitButton),
+    })
+    .click()
+}
```
[the remark this answers is not in the document]
Comment (veronikasif) [reply apps/system-e2e/src/tests/islandis/application-system/acceptance/social-insurance-administration/shared.ts:136]: The heading in `submitApplication` is different than other headings, ok as is

**Review by ylfahfa — APPROVED** [review]
Nice :cowboy_hat_face:

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]

**Pull request merged**